Sming esp-idf esp32 toolchain path

On my freshly installed Devuan machine, I found that the ```export.sh``` script doesn't add the xtensa's path ```/esp32/tools/xtensa-esp32-elf/esp-2021r1-8.4.0/xtensa-esp32-elf/bin/```.  The ```install.sh``` script exits normally.

My app builds normally once I add the path above to the PATH.

Good morning!

Forgive my lack of correctness in toolchain/compiler wording.

When I have manually added the compiler path (not to the login scripts, but by hand) and I have built some app, after logout/login Sming successfully compiles the app again. So maybe on the first build the magic about the compiler path happens.

Therefore I completely wiped esp32 and Sming from the disk and did it again. Here is the output:

```
$ make list-config 

fw: Invoking 'list-config' for Esp32/esp32 (debug) architecture
Executing action: reconfigure
Running cmake in directory /home/opt/Sming/Sming/out/Esp32/esp32/debug/build/esp32/sdk
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -DCCACHE_ENABLE=0 /home/opt/Sming/Sming/Arch/Esp32/Components/esp32/project/esp32/debug"...
-- Found Git: /usr/bin/git (found version "2.30.2") 
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- The ASM compiler identification is unknown
-- Found assembler: xtensa-esp32-elf-gcc
CMake Error at /opt/esp-idf/tools/cmake/project.cmake:283 (__project):
  The CMAKE_C_COMPILER:

    xtensa-esp32-elf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)


CMake Error at /opt/esp-idf/tools/cmake/project.cmake:283 (__project):
  The CMAKE_CXX_COMPILER:

    xtensa-esp32-elf-g++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)


CMake Error at /opt/esp-idf/tools/cmake/project.cmake:283 (__project):
  The CMAKE_ASM_COMPILER:

    xtensa-esp32-elf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "ASM" or the CMake cache entry CMAKE_ASM_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)


-- Warning: Did not find file Compiler/-ASM
-- Configuring incomplete, errors occurred!
See also "/home/opt/Sming/Sming/out/Esp32/esp32/debug/build/esp32/sdk/CMakeFiles/CMakeOutput.log".
See also "/home/opt/Sming/Sming/out/Esp32/esp32/debug/build/esp32/sdk/CMakeFiles/CMakeError.log".
cmake failed with exit code 1

** Sming build configuration **

PREREQUISITES:
CUSTOM_TARGETS:
- check-ssl
- out/Esp32/esp32/debug/firmware/spiff_rom.bin
- partmap-build
LIBS:
- app_update
- bootloader_support
- bt
- btdm_app
- c
- coexist
- core
- cxx
- driver
- efuse
- esp32
- esp_adc_cal
- esp_common
- esp_eth
- esp_event
- esp_gdbstub
- esp_hw_support
- esp_ipc
- esp_netif
- esp_pm
- esp_ringbuf
- esp_rom
- esp_system
- esp_timer
- esp_wifi
- espcoredump
- espnow
- freertos
- gcc
- hal
- heap
- log
- lwip
- m
- mbedcrypto
- mesh
- net80211
- newlib
- nvs_flash
- openssl
- phy
- pp
- pthread
- rtc
- smartconfig
- soc
- spi_flash
- stdc++
- wifi_provisioning
- wpa_supplicant
- xt_hal
- xtensa
ARDUINO_LIBRARIES:
- ArduinoJson6
- DS3232RTC
- SDCard
- TM1637
SUBMODULES:
- /home/opt/Sming/Sming/Components/FlashString
- /home/opt/Sming/Sming/Components/IFS
- /home/opt/Sming/Sming/Components/esptool/esptool
- /home/opt/Sming/Sming/Components/http-parser
- /home/opt/Sming/Sming/Components/libyuarel
- /home/opt/Sming/Sming/Components/mqtt-codec
- /home/opt/Sming/Sming/Components/ws_parser
- /home/opt/Sming/Sming/Libraries/ArduinoJson6/ArduinoJson
- /home/opt/Sming/Sming/Libraries/Spiffs/spiffs
- /home/opt/Sming/Sming/Libraries/TM1637
CONFIG_VARS:
- COM_SPEED_SERIAL = $(COM_SPEED)
- DEBUG_PRINT_FILENAME_AND_LINE = 0
- DEBUG_VERBOSE_LEVEL = 1
- ENABLE_CMD_EXECUTOR = 1
- ENABLE_CUSTOM_LWIP = 
- ENABLE_GDB = 
- ENABLE_SMART_CONFIG = 1
- ENABLE_SPI_DEBUG = 0
- ENABLE_SSL = 0
- ENABLE_TASK_COUNT = 
- ENABLE_WPS = 
- HTTP_SERVER_EXPOSE_NAME = 1
- HTTP_SERVER_EXPOSE_VERSION = 0
- HWCONFIG = spiffs-two-roms
- HWCONFIG_OPTS = 
- LOCALE = 
- RBOOT_TWO_ROMS = 1
- STRING_OBJECT_SIZE = 12
- TASK_QUEUE_LENGTH = 10
- WIFI_PWD = 
- WIFI_SSID = 
CACHE_VARS:
- APP_NAME = app
- COM_OPTS = --raw --encoding ascii --rts 0 --dtr 0
- COM_PORT = /dev/ttyUSB3
- COM_PORT_ESPTOOL = $(COM_PORT)
- COM_PORT_GDB = $(COM_PORT)
- COM_SPEED = 115200
- COM_SPEED_ESPTOOL = $(COM_SPEED)
- FULL_COMPONENT_BUILD = 
- GDB_CMDLINE = trap '' INT; $(GDB) -x $(GDBSTUB_DIR)/gdbcmds -b $(COM_SPEED_GDB) -ex "target remote $(COM_PORT_GDB)" $(TARGET_OUT)
- KILL_TERM = pkill -9 -f "$(COM_PORT) $(COM_SPEED_SERIAL)" || exit 0
- PIP_ARGS = 
- SDK_CUSTOM_CONFIG = 
- SERVER_OTA_PORT = 9999
- SPIFF_BIN = spiff_rom
- SPIFF_FILES = files
- TERMINAL = $(PYTHON) -m serial.tools.miniterm $(COM_OPTS) $(COM_PORT) $(COM_SPEED_SERIAL)
- TRACE = 
RELINK_VARS:
- DISABLE_NETWORK = 0
- DISABLE_WIFI = 0
- ENABLE_SSL = 0
- PARTITION_TABLE_OFFSET = 0x00008000
- RBOOT_RTC_ENABLED = 
- SPIFFS_OBJ_META_LEN = 16
- SPIFF_FILEDESC_COUNT = 7
- SPISOFT_DELAY_FIXED = 0
- SPISOFT_DELAY_VARIABLE = 0
- SSL_DEBUG = 0
DEBUG_VARS:
- APP_CFLAGS = -DRBOOT_TWO_ROMS=$(RBOOT_TWO_ROMS) -DCOM_SPEED_SERIAL=$(COM_SPEED_SERIAL) -DSERIAL_BAUD_RATE=$(COM_SPEED_SERIAL)
- APP_LIBDIR = out/Esp32/esp32/debug/lib
- AR = /opt/esp32/tools/xtensa-esp32-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-ar
- ARCH_BASE = /home/opt/Sming/Sming/Arch/Esp32
- AS = /opt/esp32/tools/xtensa-esp32-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
- AWK = POSIXLY_CORRECT= awk
- BLANK_BIN = /home/opt/Sming/Sming/Components/Storage/blank.bin
- BUILD_BASE = $(OUT_BASE)/build
- CC = /opt/esp32/tools/xtensa-esp32-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
- CLANG_FORMAT = clang-format
- CMAKE = cmake
- COM_SPEED_GDB = $(COM_SPEED_SERIAL)
- CXX = /opt/esp32/tools/xtensa-esp32-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-g++
- ESP32_COMPILER_PATH = /opt/esp32/tools/xtensa-esp32-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32-elf
- ESP32_OPENOCD_PATH = /opt/esp32/tools/openocd-esp32/v0.11.0-esp32-20211220
- ESP32_PYTHON_PATH = /usr/bin
- ESP32_ULP_PATH = /opt/esp32/tools/esp32ulp-elf/2.28.51-esp-20191205
- ESPTOOL = /home/opt/Sming/Sming/Components/esptool/esptool/esptool.py
- FW_BASE = $(OUT_BASE)/firmware
- GCC_VERSION = 8.4.0
- GDB = /opt/esp32/tools/xtensa-esp32-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gdb
- GDBSTUB_DIR = /home/opt/Sming/Sming/Arch/Esp32/Components/gdbstub
- GIT = git
- GLOBAL_CFLAGS = -DSMING_ARCH=$(SMING_ARCH) -DSMING_SOC=$(SMING_SOC) -DPROJECT_DIR=\"$(PROJECT_DIR)\" -DSMING_HOME=\"$(SMING_HOME)\" $(USER_CFLAGS) -D$(SMING_SOC_VAR)=1 -DENABLE_CMD_EXECUTOR=$(ENABLE_CMD_EXECUTOR) -DDEBUG_PRINT_FILENAME_AND_LINE=$(DEBUG_PRINT_FILENAME_AND_LINE) -DCUST_FILE_BASE=$$* -DDEBUG_VERBOSE_LEVEL=$(DEBUG_VERBOSE_LEVEL) -DSTRING_OBJECT_SIZE=$(STRING_OBJECT_SIZE)  -DENABLE_SMART_CONFIG=1 -DHTTP_SERVER_EXPOSE_NAME=$(HTTP_SERVER_EXPOSE_NAME) -DHTTP_SERVER_EXPOSE_VERSION=$(HTTP_SERVER_EXPOSE_VERSION)
- IDF_PATH = /opt/esp-idf
- IDF_TOOLS_PATH = /opt/esp32
- IDF_VER = v4.3-1675-g4a43d72911
- LD = /opt/esp32/tools/xtensa-esp32-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
- NINJA = ninja
- NM = /opt/esp32/tools/xtensa-esp32-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-nm
- OBJCOPY = /opt/esp32/tools/xtensa-esp32-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-objcopy
- OBJDUMP = /opt/esp32/tools/xtensa-esp32-elf/esp-2021r2-patch3-8.4.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-objdump
- OS = 
- OUT_BASE = out/Esp32/esp32/debug
- PROJECT_DIR = /home/mzk/esp/pt32/fw
- PROJECT_SOC = esp32 esp32c3 esp32s2 esp32s3 esp8266
- PYTHON = /usr/bin/python3
- SMING_ARCH = Esp32
- SMING_ARCH_FULL = Esp32/esp32
- SMING_CXX_STD = c++17
- SMING_C_STD = c11
- SMING_HOME = /home/opt/Sming/Sming
- SMING_RELEASE = 
- SMING_SOC = esp32
- SPIFF_BIN_OUT = out/Esp32/esp32/debug/firmware/spiff_rom.bin
- SPI_MODE = $(STORAGE_DEVICE_spiFlash_MODE)
- SPI_SIZE = $(STORAGE_DEVICE_spiFlash_SIZE)
- SPI_SPEED = $(STORAGE_DEVICE_spiFlash_SPEED)
- STRICT = 0
- TARGET_BIN = $(FW_BASE)/$(APP_NAME).bin
- TARGET_OUT = $(BUILD_BASE)/$(APP_NAME).out
- TARGET_OUT_0 = $(TARGET_OUT) 
- TOOLS_BASE = $(SMING_HOME)/$(OUT_BASE)/tools
- UNAME = Linux
- USER_LIBDIR = $(SMING_HOME)/$(OUT_BASE)/lib


make: Nothing to be done for 'list-config'.
```

And the output when I try to ```make``` my app:
```
make

fw: Invoking 'all' for Esp32/esp32 (debug) architecture
Executing action: reconfigure
Running cmake in directory /home/opt/Sming/Sming/out/Esp32/esp32/debug/build/esp32/sdk
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -DCCACHE_ENABLE=0 /home/opt/Sming/Sming/Arch/Esp32/Components/esp32/project/esp32/debug"...
-- Found Git: /usr/bin/git (found version "2.30.2") 
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- The ASM compiler identification is unknown
-- Found assembler: xtensa-esp32-elf-gcc
CMake Error at /opt/esp-idf/tools/cmake/project.cmake:283 (__project):
  The CMAKE_C_COMPILER:

    xtensa-esp32-elf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)


CMake Error at /opt/esp-idf/tools/cmake/project.cmake:283 (__project):
  The CMAKE_CXX_COMPILER:

    xtensa-esp32-elf-g++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)


CMake Error at /opt/esp-idf/tools/cmake/project.cmake:283 (__project):
  The CMAKE_ASM_COMPILER:

    xtensa-esp32-elf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "ASM" or the CMake cache entry CMAKE_ASM_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)


-- Warning: Did not find file Compiler/-ASM
-- Configuring incomplete, errors occurred!
See also "/home/opt/Sming/Sming/out/Esp32/esp32/debug/build/esp32/sdk/CMakeFiles/CMakeOutput.log".
See also "/home/opt/Sming/Sming/out/Esp32/esp32/debug/build/esp32/sdk/CMakeFiles/CMakeError.log".
cmake failed with exit code 1
Executing action: reconfigure
Running cmake in directory /home/opt/Sming/Sming/out/Esp32/esp32/debug/build/esp32/sdk
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -DCCACHE_ENABLE=0 /home/opt/Sming/Sming/Arch/Esp32/Components/esp32/project/esp32/debug"...
-- Found Git: /usr/bin/git (found version "2.30.2") 
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- The ASM compiler identification is unknown
-- Found assembler: xtensa-esp32-elf-gcc
CMake Error at /opt/esp-idf/tools/cmake/project.cmake:283 (__project):
  The CMAKE_C_COMPILER:

    xtensa-esp32-elf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)


CMake Error at /opt/esp-idf/tools/cmake/project.cmake:283 (__project):
  The CMAKE_CXX_COMPILER:

    xtensa-esp32-elf-g++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)


CMake Error at /opt/esp-idf/tools/cmake/project.cmake:283 (__project):
  The CMAKE_ASM_COMPILER:

    xtensa-esp32-elf-gcc

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "ASM" or the CMake cache entry CMAKE_ASM_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)


-- Warning: Did not find file Compiler/-ASM
-- Configuring incomplete, errors occurred!
See also "/home/opt/Sming/Sming/out/Esp32/esp32/debug/build/esp32/sdk/CMakeFiles/CMakeOutput.log".
See also "/home/opt/Sming/Sming/out/Esp32/esp32/debug/build/esp32/sdk/CMakeFiles/CMakeError.log".
cmake failed with exit code 1
make: *** [/home/opt/Sming/Sming/Arch/Esp32/Components/esp32/component.mk:311: /home/opt/Sming/Sming/out/Esp32/esp32/debug/build/esp32/sdk/config/sdkconfig.h] Error 2
```

--- a/Sming/Arch/Esp32/build.mk
+++ b/Sming/Arch/Esp32/build.mk
@@ -100,8 +100,8 @@ endif
 DEBUG_VARS += NINJA
 NINJA := $(if $(ESP32_NINJA_PATH),$(ESP32_NINJA_PATH)/,)ninja
 
-space :=
-space +=
+empty:=
+space:= $(empty) $(empty)
 
 export PATH := $(subst $(space),:,$(IDF_PATH_LIST)):$(PATH)
 

